include/caffe/net.hpp

```diff
@@ -192,7 +192,8 @@ class Net {
   /// @brief Append a new bottom blob to the net.
   int AppendBottom(const NetParameter& param, const int layer_id,
                    const int bottom_id, set<string>* available_blobs,
-                   map<string, int>* blob_name_to_idx);
+                   map<string, int>* blob_name_to_idx,
+                   bool skip_propagate = false);
```

Comment:
Leaving the check in `Net::Init` is fine (though I'd also be fine with moving it to `AppendBottom`).  The rest of the logic is more natural to move to `AppendBottom` (and its signature shouldn't change) since it's associated specifically with a new bottom blob, and `Net::Init` is already longer than a single method should be.